pkgs/servers/nosql/mongodb/v4_0.nix

```diff
@@ -11,7 +11,11 @@ let
 in buildMongoDB {
   version = "4.0.27";
   sha256 = "sha256-ct33mnK4pszhYM4Is7j0GZQRyi8i8Qmy0wcklyq5LjM=";
-  patches =
-    [ ./forget-build-dependencies.patch ./mozjs-45_fix-3-byte-opcode.patch ]
+  patches = [
+    ./forget-build-dependencies.patch
+    ./mozjs-45_fix-3-byte-opcode.patch
+    ./mongodb-4.4.1-gcc11.patch # https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-db/mongodb/files/mongodb-4.4.1-gcc11.patch
```

Comment:
Are we sure gentoo provides a stable URL? I don't see any hash or version in it. Might be more robust to just keep the copy here.

Reply:
https://gitweb.gentoo.org/repo/gentoo.git/commit/dev-db/mongodb/files/mongodb-4.4.1-gcc11.patch?id=9bcbb4f7a076e7839c55678123c67a1c7b838237

Here you go.

Reply:
I think there is a mirror on Github, but also vote to keep a copy here, another patch is a slighy modified version of ther patch, so we just organize what we need.

Also we probably would need to combo patches + fetchpatch :P

Reply:
Either the GitHub mirror or this [URL](https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-db/mongodb/files/mongodb-4.4.1-gcc11.patch?id=9bcbb4f7a076e7839c55678123c67a1c7b838237) should be fine.

> Also, we probably would need to combo patches + fetchpatch :package: 

I see no problem in this, this also happen in other packages.

Reply:
Note, I am not completely against this approach, but using `fetchpatch` will make future bumps easier I think (instead of copying the contents of the patch in case of changes, we can just update the URL+hash). Also, it gives it more context.

Reply:
Done.